com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -561,6 +561,9 @@ internal void CleanDiffedSceneObjects()
                 {
                     UnityEngine.Object.Destroy(networkObjectsToDestroy[i].gameObject);
                 }
+
+                // Make sure to clear this once done destroying all remaining NetworkObjects
+                PendingSoftSyncObjects.Clear();
```

Comment:
looking at the code above, shouldn't we just iterate over `PendingSoftSyncObjects`, call `UnityEngine.Object.Destroy` on them and then do `PendingSoftSyncObjects.Clear()`?
I think we don't need `networkObjectsToDestroy` at all.

I'd suggest something like this (pseudo-code):

```cs
// Clean up any in-scene objects that had been destroyed
if (PendingSoftSyncObjects.Count > 0)
{
    foreach (var pair in PendingSoftSyncObjects)
    {
        UnityEngine.Object.Destroy(pair.Value.gameObject);
    }

    // Make sure to clear this once done destroying all remaining NetworkObjects
    PendingSoftSyncObjects.Clear();
}
```

Reply:
Yeah that should work since it doesn't modify the actual dictionary but just makes the ghost Unity References null. Either approach works, this one being cleaner.

Reply:
Just applied your suggested fix @MFatihMAR